lib/blacklight/document/cache_key.rb

```diff
@@ -0,0 +1,28 @@
+# -*- encoding : utf-8 -*-
+# This module provides the cache key which can be used by rails
+# caching to determine when to expire a particular object's cache
+# See http://apidock.com/rails/ActiveRecord/Integration/cache_key
+# This key should be used in conjunction with additional data to
+# determine when a document can be cached (e.g. for different view
+# types in search results like gallery and list)
+module Blacklight::Document::CacheKey
+  def cache_key
+    case
+    when new_record?
+      "#{self.class.model_name.cache_key}/new"
+    when timestamp = self[timestamp_key]
+      timestamp = Time.zone.parse(timestamp).utc.to_s(cache_timestamp_format)
+      "#{self.class.model_name.cache_key}/#{id}-#{timestamp}"
+    else
+      "#{self.class.model_name.cache_key}/#{id}"
+    end
+  end
+
+  def timestamp_key
+    :last_updated
```

Comment:
I'm not sure we should tie this directly to timestamps; I can imagine using `__version__` (baking into Solr 5.x, and a requirement for SolrCloud in 4.x) here instead.